### PR TITLE
production configuration (optional)

### PR DESCRIPTION
**Pull request by vtuson**
**What this PR does / why we need it**:
Added two things here:
* an alternative (optional) values-production yaml which is opinionated on how to set up the chart for production environment (using ingress, clusterIp, RWX PVC,..)
* comments on the README to explain how to horizontally scale the chart and how to use nfs-server-provisioner for it.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by unguiculus** [comment]
A separate values file doesn't make sense. You can't use it anyways when you install the chart from the repo. That's rather something that should go into the readme.

**Comment by vtuson** [comment]
@unguiculus  it is linked from the readme, dont you think it would make a very long readme to have a whole values file on it?

File: stable/wordpress/README.md [hunk 1]
```diff
@@ -116,6 +116,23 @@ $ helm install --name my-release -f values.yaml stable/wordpress
 
 > **Tip**: You can use the default [values.yaml](values.yaml)
 
+## Production and horizontal scaling
+
+The following repo contains the recommended production settings for wordpress capture in an alternative [values file](values-production.yaml). Please read carefully the comments in the values-production.yaml file to set up your environment appropiately.
```
Comment (sameersbn) [inline stable/wordpress/README.md]: typo `appropiately`

File: stable/wordpress/values-production.yaml [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    ##   GKE, AWS & OpenStack)
+    ##
+    # storageClass: "-"
+    accessMode: ReadWriteOnce
+    size: 8Gi
+
+## Kubernetes configuration
+## For minikube, set this to NodePort, elsewhere use LoadBalancer or ClusterIP
+##
+serviceType: ClusterIp
```
Comment (sameersbn) [inline stable/wordpress/values-production.yaml]: -> `ClusterIP`

**Review by sameersbn — CHANGES_REQUESTED** [review]

**Comment by sameersbn** [comment]
@unguiculus: @vtuson is providing a sane default configuration for production use cases with scalability and high availability in mind. As you rightly pointed out, the `values-production.yaml` file will not be available directly while installing the chart from a chart repo and perhaps should also be ignored (.helmignore) while packaging the chart. However I think the appropriate place for this configuration is the the git repo of the chart. 

@vtuson could you add `values-production.yaml` to the `.helmignore` file?

**Comment by vtuson** [comment]
@sameersbn done

File: stable/wordpress/README.md (hunk 1 above)
Comment (vtuson) [reply stable/wordpress/README.md]: done

File: stable/wordpress/values-production.yaml (hunk 2 above)
Comment (vtuson) [reply stable/wordpress/values-production.yaml]: done

**Review by sameersbn — APPROVED** [review]

**Comment by sameersbn** [comment]
@unguiculus are you okay with these changes being merged? do you have any other concerns with the addition of a values-production.yaml to the chart directory?

File: stable/wordpress/README.md [hunk 3]
```diff
@@ -116,6 +116,23 @@ $ helm install --name my-release -f values.yaml stable/wordpress
...
+
+```console
+$ helm install --name my-release -f values-production.yaml stable/wordpress
+$ kubectl scale deployment my-wp-deployment --replicas=3 
+```
+To use the /admin portal and to ensure you can scale wordpress you need to provide a ReadWriteMany PVC, if you dont have a provisioner for this type of storage, we recommend that you install the nfs provisioner and map it to a RWO volume.
+
+```console
+$ helm install stable/nfs-server-provisioner --set persistence.enabled=true,persistence.size=10Gi
+$ helm install --name my-release -f values-production.yaml --set persitence.storageClass=nfs stable/wordpress
```
Comment (unguiculus) [inline stable/wordpress/README.md:134]: This won't work as is. The docs should at least mention that `values-production.yaml` has to be copied to a local directory.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/wordpress/README.md (hunk 3 above)
Comment (vtuson) [reply stable/wordpress/README.md:134]: will do

File: stable/wordpress/README.md [hunk 4]
```diff
@@ -120,10 +120,11 @@ $ helm install --name my-release -f values.yaml stable/wordpress
 
 The following repo contains the recommended production settings for wordpress capture in an alternative [values file](values-production.yaml). Please read carefully the comments in the values-production.yaml file to set up your environment appropriately.
 
-To horizontally scale this chart:
+To horizontally scale this chart, first download the values-production.yaml file to your local folder, then:
```
Comment (sameersbn) [inline stable/wordpress/README.md]: Would be nice if we hotlink to the file in the repo, `[values-production.yaml](values-production.yaml)`

Comment (vtuson) [reply stable/wordpress/README.md]: it is hotlink from above.. I would hot link it again.

Comment (vtuson) [reply stable/wordpress/README.md]: done

Comment (sameersbn) [reply stable/wordpress/README.md]: thanks @vtuson 
@unguiculus could you please review

**Pull request merged**